airflow/www/utils.py

```diff
@@ -76,6 +76,121 @@ def is_accessible(self):
+    """
+
+    def get_params(**kwargs):
+        params = []
+        for k, v in kwargs.items():
+            if k == 'showPaused' and v:
+                continue
+            if v:
+                params.append('{}={}'.format(k, v))
+        return '&'.join(params)
+
+    void_link = 'javascript:void(0)'
+    first_node = """<li class="paginate_button {disabled}" id="dags_first">
+    <a href="{href_link}" aria-controls="dags" data-dt-idx="0" tabindex="0">&laquo;</a>
+</li>"""
+
+    previous_node = """<li class="paginate_button previous {disabled}" id="dags_previous">
+    <a href="{href_link}" aria-controls="dags" data-dt-idx="0" tabindex="0">&lt;</a>
+</li>"""
+
+    next_node = """<li class="paginate_button next {disabled}" id="dags_next">
+    <a href="{href_link}" aria-controls="dags" data-dt-idx="3" tabindex="0">&gt;</a>
+</li>"""
+
+    last_node = """<li class="paginate_button {disabled}" id="dags_last">
+    <a href="{href_link}" aria-controls="dags" data-dt-idx="3" tabindex="0">&raquo;</a>
+</li>"""
+
+    page_node = """<li class="paginate_button {is_active}">
+    <a href="{href_link}" aria-controls="dags" data-dt-idx="2" tabindex="0">{page_num}</a>
+</li>"""
+
+    output = ['<ul class="pagination" style="margin-top:0px;">']
+
+    is_disabled = 'disabled' if current_page <= 0 else ''
+    output.append(first_node.format(href_link="?{}"
+                                    .format(get_params(page=0,
+                                                       search=search,
+                                                       showPaused=showPaused)),
+                                    disabled=is_disabled))
+
+    page_link = (void_link
+                 if current_page <= 0
+                 else ('?{}'.format(get_params(page=(current_page - 1),
+                                               search=search,
+                                               showPaused=showPaused))
+                       if current_page > 0 else '/'))
+    output.append(previous_node.format(href_link=page_link,
+                                       disabled=is_disabled))
+
+    mid = int(window / 2)
+    last_page = num_of_pages - 1
+
+    if current_page <= mid or num_of_pages < window:
+        pages = [i for i in range(0, min(num_of_pages, window))]
+    elif mid < current_page < last_page - mid:
+        pages = [i for i in range(current_page - mid, current_page + mid + 1)]
+    else:
+        pages = [i for i in range(num_of_pages - window, last_page + 1)]
+
+    def is_current(current, page):
+        return page == current
+
+    for page in pages:
+        vals = {
+            'is_active': 'active' if is_current(current_page, page) else '',
+            'href_link': void_link if is_current(current_page, page)
+                         else '?{}'.format(get_params(page=page,
+                                                      search=search,
+                                                      showPaused=showPaused)),
```

Comment:
Does showPaused get propagated correctly?  Seems like it might not be consumed correctly. If it seems to render correctly I probably just misread something.

Reply:
Fixed.